lib/puppet/file_system/uniquefile.rb

```diff
@@ -131,6 +131,9 @@ def create_tmpname(basename, *rest)
     n = nil
     begin
       path = File.expand_path(make_tmpname(basename, n), tmpdir)
+      if Puppet.features.microsoft_windows? # Prefixing the path with \\?\ lets us use more than 260 characters on Windows.
+        path = "\\\\?\\#{path}".gsub('/', '\\')
```

Comment:
Rather than address here, can you just put this into the tests for now when the temp paths are constructed?

Reply:
Unfortunately not. We could configure the bucketdir to have the `\\?\` prefix, but the `gsub` has to be done on the final path to convert it to a proper Windows path for it to work, which means we need to hook in to the internals of `P::FS::Uniquefile`.